Mathlib/Analysis/Calculus/FDeriv/Partial.lean

```diff
@@ -0,0 +1,334 @@
+/-
+Copyright (c) 2025 Igor Khavkine. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Igor Khavkine
+-/
+import Mathlib.Analysis.Calculus.MeanValue
+
+/-!
+# Partial derivatives
+
+In this file we prove some basic facts about partial derivatives of functions
+defined on a product space, like `f : E × F → G`:
+
+* `HasFDerivWithinAt.partial_fst` , `HasFDerivWithinAt.partial_snd`: if `f` is differentiable
+   with derivative `f' z` at `z`, then the partial derivatives of `(f ∘ (z.1, ·))`
+   and `(f ∘ (·, z.2))` are respectively `(f' z) ∘L (.inl 𝕜 E F)` and
+   `(f' z) ∘L (.inr 𝕜 E F)`. If `f'` is continuous, then continuity can be obtained by
+   by combining `Continuous(|At|On|WithinAt).clm_comp` and `Continuous(|At|On|WithinAt)_const`.
+
+* `hasFDerivWithinAt_of_partial_snd_continuousWithinAt_prod_open` : a weak sufficient condition
+  for differeniability of `f` at `z = (x,y)` is that, say, the first derivative (within set `s`)
+  `f'xz` exists at `z`, while the second partial derivative `f'y z` exists and is jointly
+  continuous at `z` in the product set `s ×ˢ t` where `t` is open, with the derivative given by
+  `f'z = f'xz.coprod (f'y z)`. `hasFDerivWithinAt_of_partial_fst_continuousWithinAt_prod_open` has
+  the roles of the partial derivatives reversed.
+
+  The proofs follow §9.8.1 from Dieudonné's *Foundations of Modern Analysis* (1969).
```

Comment:
Hi again! I found the 1960 edition in the library and in that this result is at 8.9.1 not 9.8.1. In any case I'm pretty sure you could find a shorter proof of `hasFDerivWithinAt_of_partial_snd_continuousWithinAt_prod_open` if you felt able to disregard Dieudonné and instead begin from `hasFDerivWithinAt_uncurry_of_continuousWithinAt_snd` in [this](https://github.com/agjftucker/mathlib4/tree/agjftucker/implicit) WIP branch.

Reply:
> I think some of the proofs below the four newly introduced theorems could be made shorter by using them

After reorganisation, these four have become three at the top of Calculus.FDeriv.Partial and two at the bottom of Normed.Module.Convex.